src/NavItem.js

```diff
@@ -28,7 +28,7 @@ class NavItem extends React.Component {
   }
 
   handleClick(e) {
-    if (this.props.onSelect) {
+    if (this.props.onSelect && this.props.onSelect.name !== 'bound handleCollapse') {
```

Comment:
when using onSelect, it's stopping the default browser behavior (if any) and runs only the custom function.